drivers/spi/spi_nrfx_spim.c

```diff
@@ -765,6 +765,28 @@ static int spi_nrfx_init(const struct device *dev)
 #endif
 	return pm_device_driver_init(dev, spim_nrfx_pm_action);
 }
+
+static int spi_nrfx_deinit(const struct device *dev)
+{
+#if defined(CONFIG_PM_DEVICE)
+	enum pm_device_state state;
+
+	/*
+	 * PM must have suspended the device before driver can
+	 * be deinitialized
+	 */
+	(void)pm_device_state_get(dev, &state);
+	return state == PM_DEVICE_STATE_SUSPENDED ||
+	       state == PM_DEVICE_STATE_OFF ?
+	       0 : -EBUSY;
+#else
+	/* PM suspend implementation does everything we need */
+	spim_suspend(dev);
+#endif
```

Comment:
not related to this PRis this supposed to be improved by the pm_device_deinit you are adding on another PR? or does it just become something like
#ifdef CONFIG_PM_DEVICE
pm_device_deinit(..)
#else
spim_suspend
#endif

Because this is what I would like to avoid happening on drivers, is always having two different implementations of the same function depending on PM_DEVICE on or off